### PR TITLE
feat - added products and filter functionality in product page

### PR DESCRIPTION
**Pull request by Balakumar3018**
Ignore responsiveness
1. Need to refactor code before submission.
Preview -  https://marvelous-liger-7b9181.netlify.app/products

### CODE REVIEW

File: src/components/ProductList/ProductList.css [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+.btn{
+    padding: 4px;
+}
+.product-details{
+    text-align: left;
+    padding-left: 16px;
+}
+.basic-card{
+    border:"1px solid black";
+    margin:8px;
```
Comment (vikrantkode) [inline src/components/ProductList/ProductList.css:22]: use variable for naming colors

File: src/components/ProductList/ProductList.jsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            <div className="basic-card " key={item.productName}> 
+                <img src={item.displayImg} width="100%" height="270px" alt={item.productName} />
+                <div className="product-details">
+                    <div className="product-name">{item.productName}</div>
+                    <div>RAM: {item.RAM}</div>
+                    <div>Price: {item.price}</div>
+                    <div>Rating: {item.rating}</div>
+                    {/* {item.inStock && <div >Instock</div>} */}
+                    {/* {!item.inStock && <div>Out of stock</div>} */}
+                </div>
```
Comment (vikrantkode) [inline src/components/ProductList/ProductList.jsx:17]: remove commented code

File: src/pages/products-page/ProductPage.jsx [hunk 3]
```diff
@@ -1,5 +1,162 @@
...
+                    return {...state, rating:action.payload} 
+                case "category":
+                    return {...state, Showcategory:action.payload}    
+                case "clear":
+                    return {
+                        sortby:"null",
+                        rating:"null"
+                    }      
+                default:
+                    return state;
```
Comment (vikrantkode) [inline src/pages/products-page/ProductPage.jsx]: u can separate out reducer function for better code structure

**Review by vikrantkode — APPROVED** [review]
in UI - Shop now btn is not working on hero image  and button styling is not same for this
- Some changes suggested .. you can have a look on them

**Comment by Balakumar3018** [comment]
sure will add suggestions thanks

File: src/pages/products-page/ProductPage.jsx [hunk 4]
```diff
@@ -1,5 +1,162 @@
...
+        useEffect(() => {
+            const fetchProducts = async () => {
+              const res = await axios.get("/api/products");
+              setProducts(res.data.products);
+            };
+            fetchProducts();
+          }, []);
+          const ProductReducerFunction=(state,action)=>{
+            switch(action.type){
+                case "sort":
```
Comment (arulk8) [inline src/pages/products-page/ProductPage.jsx]: this "sort" can be moved to separate constants file and that can be used in other places

**Pull request merged**